macros/src/codegen.rs

```diff
@@ -293,21 +312,16 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
 
                 let state_data = match sm.state_data.data_types.get(state) {
                     Some(st) => {
-                        quote! { state_data: &#st, }
+                        quote! { state_data: #st, }
                     }
                     None => {
                         quote! {}
                     }
                 };
                 let event_data = match sm.event_data.data_types.get(event) {
-                    Some(et) => match et {
-                        Type::Reference(_) => {
-                            quote! { event_data: #et }
-                        }
-                        _ => {
-                            quote! { event_data: &#et }
-                        }
-                    },
```

Comment:
Why are we losing the ability to hold references as the event data? Seems like we could still support this.

Reply:
We are not losing this ability, because in actions we are using the type as it was provided by the user.

So a `MyEventData` will stay `MyEventData`, a `&MyEventData` will stay `&MyEventData`, and a `&mut MyEventData` will stay `&mut MyEventData`.

`event_with_mutable_data` expands to:

```rust
pub trait StateMachineContext {
    #[allow(missing_docs)]
    fn guard<'a>(&mut self, event_data: &'a mut MyEventData) -> Result<(), ()>;
    #[allow(missing_docs)]
    fn action<'a>(&mut self, event_data: &'a mut MyEventData) -> ();
}
```

Reply:
Ah, you're indeed correct. Sorry!